onnxruntime/core/mlas/lib/pooling.cpp

```diff
@@ -1224,7 +1224,7 @@ Return Value:
         InputSize *= WorkBlock.InputShape[dim];
         OutputSize *= WorkBlock.OutputShape[dim];
 
-        InputAndKernelShapeMatch &= (WorkBlock.KernelShape[dim] == int64_t(WorkBlock.InputShape[dim]));
+        InputAndKernelShapeMatch &= (dim < 3 && WorkBlock.KernelShape[dim] == int64_t(WorkBlock.InputShape[dim]));
```

Comment:
What was the xcode warning here? Seems like an odd change given the surrounding code is using dim to index the work block too.

Reply:
It complains something like index out of array bound.

Reply:
But it no longer complains about the other parts of the function? It's weird because WorkBlock.InputShape[dim] is used right above and WorkBlock.KernelShape[dim] is used below. I hate using exceptions, but if you add something like the below to the top, does that mask the warning? Some other code analysis tool complained about a different part of this function thinking that Dimensions could be any value.

if (Dimensions == 0 || Dimensions > 3) {
    throw std::invalid_argument("bad dimensions");
}

Reply:
No it did not complain on other parts only this line - the change could silence the alert proved.

Reply:
if (Dimensions == 0 || Dimensions > 3) {
throw std::invalid_argument("bad dimensions");
} 
Cause the UT fail.